src/app/cli/src/commands/verify_command.rs

```diff
@@ -73,27 +79,73 @@ impl VerifyCommand {
 
     async fn verify(
         &self,
-        options: VerificationOptions,
+        options: Arc<VerificationOptions>,
         listener: Option<Arc<VerificationMultiProgress>>,
     ) -> GenericVerificationResult {
-        let dataset_handle = self
+        let dataset_ref_arg = self.refs.first().unwrap();
+        let dataset_ref_pattern = DatasetRefPattern {
+            pattern: dataset_ref_arg.to_string(),
+            ..DatasetRefPattern::default()
+        };
+        if !dataset_ref_arg
+            .to_string()
+            .contains(dataset_ref_pattern.wildcard)
```

Comment:
Why would this need a separate algorithm? A wildcard without instance of `%` will return the same results as listing all datasets.

Reply:
One more thing:
 - in wildcard mode this is okay to say if pattern did not match anything - this needs a warning though
 - in non-wildcard mode, we should still have an error saying dataset is not found.